crypto/src/main/java/org/cryptimeleon/incentive/crypto/IssueJoinRandomness.java

```diff
@@ -15,4 +19,9 @@ public class IssueJoinRandomness {
     final Zn.ZnElement t;
     final Zn.ZnElement u;
     final Zn.ZnElement blindGenesisR;
+
+    static IssueJoinRandomness generate(IncentivePublicParameters pp) {
+        Zn.ZnElement[] rv = Stream.generate(pp.getBg().getZn()::getUniformlyRandomElement).limit(7).toArray(Zn.ZnElement[]::new);
```

Comment:
we have a method like `zn.getUniformlyRandomElements(7)`

Reply:
I tried that, but how do I get the ZnElements from that RingElemenVector?

Reply:
```zn.getUniformlyRandomElements(7).map( ZnElement.class::cast )```
And then `.get(i)`

But I see your point. It's not incredibly elegant.

Reply:
Well it's definitely better than my version! Thanks :)